### ISSUE TITLE
Set up scaffolding

### ISSUE THREAD

**Issue by jellegerbrandy** [Paratii-Video/paratii-portal#7]
We need different directories for 

    /player
    /portal
    /assets (shared assets)

[elia does not agree with this ^^^]

We also need different `build` and `test` commands, probably along the lines of:

    yarn testPlayer   (run tests in /player)
    yarn testPortal (run tests for /portal
    yarn test (run all tests)

And, similarly:

     yarn buildPlayer
     yarn buildPortal
     yarn build

@bent0b0x , does it make sense like this?

- [ ] still need to separate build processes for embeddable player and portal

**Comment by jellegerbrandy** [comment]
@eliawk , does this look reasonable to you?

**Comment by eliawk** [comment]
😄 
my proposal for the scaffolding is more components oriented, something like
/components
/modules
/section 

![](http://bradfrost.com/wp-content/uploads/2013/06/atomic-design.png)

i also found this https://arc.js.org/

**Comment by bent0b0x** [comment]
@jellegerbrandy what you laid out makes sense. I'll try and finish up the portal scaffold today and then once I do move it into its own directory (`/portal`). 

As for `arc.js`, I think we can incorporate most of what it includes on our own, so not sure if it gives us too much benefit at this point.

**Comment by bent0b0x** [comment]
Also @eliawk I still plan on adding more structure to the `js`, including separate directories for `components` and such

**Comment by jellegerbrandy** [comment]
We still need a separate builds for player and portal.

**Comment by jellegerbrandy** [comment]
good points.  Here are some more observations:

- there is a difference bewteen "npm package" and "github repo".  t's ok to develop different packages in one repo (look wat web3 does: https://github.com/ethereum/web3.js/tree/1.0/packages)
- having different builds from the same code: we may also imagine that we have different player-builds (with or without certain functionality, say); that we have builds for different platforms, etc.
- (Not directly relevant to the discussion, but: I think the publsh-to-npm-and-then-import-again is just cumbersome and unnecessary, as we can also install packages directly from github.)
- **the bottom line, I think, is that** developing both player and portal in the same repo would not only be for sharing static assets, but also to (1) keep development in sync (i.e. not having to keep different repo's  in sync) and (2) share devpops infrastructure (test suite, CI, deployment procedure, etc.

At this point, my personal opinion tends to agree with @bent0b0x  and to have separate repositorie after all.

**Comment by jellegerbrandy** [comment]
I'm closing this for now - we can revisit this later if necessary
